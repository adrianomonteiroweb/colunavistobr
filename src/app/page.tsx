import Hero from "@/components/app/pages/hero";

const heroProps = {
  projectTitle: "Coluna Visto BR: Transformando Vidas no Brasil",
  projectDescription:
    "Levamos amor, educação e dignidade para comunidades vulneráveis. Conheça a missão de Victoria Barros e como você pode fazer a diferença.",
  founderName: "Victoria Barros",
  founderDescription: "Fundadora e Coração do Projeto",
  founderImageUrl: "/placeholder.svg?height=200&width=200",
  pixKey: "13996841291",
  pixQrCodeUrl: "/placeholder.svg?height=150&width=150",
  pixInstructions: "Sua ajuda transforma vidas!",
  instagramUrl: "https://instagram.com/colunavisto.br",
  facebookUrl: "https://facebook.com/tvcolunavisto",
  youtubeUrl: "https://youtube.com/@victoriabarros.colunavisto",
};

const Home = () => {
  return (
    <main className="min-h-screen bg-white">
      <Hero {...heroProps} />
    </main>
  );
};

export default Home;
