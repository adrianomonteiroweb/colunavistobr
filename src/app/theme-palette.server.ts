import { getThemePaletteFromSettings } from "@/lib/server-utils";
import { PALETTES, ThemePalette } from "@/components/theme-provider";

export const getInitialThemePalette = async (): Promise<ThemePalette> => {
  const palette = await getThemePaletteFromSettings();
  return palette || PALETTES[0];
};
