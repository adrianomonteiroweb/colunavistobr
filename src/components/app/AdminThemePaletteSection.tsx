"use client";
import { useTheme } from "@/components/theme-provider";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { cn } from "@/lib/utils";

const AdminThemePaletteSection = () => {
  const { currentPaletteName, setPalette, palettes } = useTheme();

  return (
    <div className="container mx-auto py-8 px-4 max-w-4xl">
      <Card className="border border-[var(--card-border)] shadow-sm bg-[var(--card-background)] mb-8">
        <CardHeader>
          <CardTitle className="text-[var(--text-primary)]">
            Escolha sua Paleta de Cores
          </CardTitle>
        </CardHeader>
        <CardContent className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
          {palettes.map((palette) => (
            <Card
              key={palette.name}
              className={cn(
                "cursor-pointer transition-all duration-200 hover:shadow-lg",
                "border-2",
                currentPaletteName === palette.name
                  ? "border-[var(--primary)] shadow-md"
                  : "border-[var(--card-border)]"
              )}
              onClick={() => setPalette(palette)}
            >
              <CardContent className="p-4 flex flex-col items-center text-center">
                <h3 className="text-lg font-semibold text-[var(--text-primary)] mb-3">
                  {palette.name}
                </h3>
                <div className="flex gap-2 mb-4">
                  <div
                    className="w-8 h-8 rounded-full border border-gray-200"
                    style={{ backgroundColor: palette.background }}
                    title="Fundo da Página"
                  />
                  <div
                    className="w-8 h-8 rounded-full border border-gray-200"
                    style={{ backgroundColor: palette.background }}
                    title="Fundo do Card"
                  />
                  <div
                    className="w-8 h-8 rounded-full border border-gray-200"
                    style={{ backgroundColor: palette.text }}
                    title="Texto Principal"
                  />
                  <div
                    className="w-8 h-8 rounded-full border border-gray-200"
                    style={{ backgroundColor: palette.primary }}
                    title="Botão Principal"
                  />
                </div>
                <Button
                  variant={
                    currentPaletteName === palette.name ? "default" : "outline"
                  }
                  className="w-full"
                  onClick={() => setPalette(palette)}
                >
                  {currentPaletteName === palette.name
                    ? "Selecionado"
                    : "Selecionar"}
                </Button>
              </CardContent>
            </Card>
          ))}
        </CardContent>
      </Card>
    </div>
  );
};

export default AdminThemePaletteSection;
