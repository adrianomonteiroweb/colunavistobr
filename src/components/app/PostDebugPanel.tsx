"use client";
import { useState } from "react";
import { Button } from "@/components/ui/button";
import { deletePost, fetchPosts } from "@/actions/postActions";

export const PostDebugPanel = () => {
  const [debugOutput, setDebugOutput] = useState<string[]>([]);
  const [loading, setLoading] = useState(false);

  const addDebugLine = (line: string) => {
    setDebugOutput((prev) => [
      ...prev,
      `${new Date().toLocaleTimeString()}: ${line}`,
    ]);
  };

  const testFetchPosts = async () => {
    setLoading(true);
    addDebugLine("🔍 Testando fetchPosts...");

    try {
      const posts = await fetchPosts();
      addDebugLine(
        `✅ fetchPosts executado com sucesso. Posts encontrados: ${posts.length}`
      );
      posts.forEach((post, index) => {
        addDebugLine(
          `  Post ${index + 1}: ID=${post.id}, Title="${post.title}"`
        );
      });
      return posts;
    } catch (error) {
      addDebugLine(`❌ Erro no fetchPosts: ${error}`);
      return [];
    } finally {
      setLoading(false);
    }
  };

  const testDeletePost = async () => {
    setLoading(true);
    addDebugLine("🗑️ Testando deletePost...");

    try {
      const posts = await fetchPosts();
      if (posts.length === 0) {
        addDebugLine("❌ Nenhum post encontrado para testar delete");
        return;
      }

      // Procurar por um post de teste que pode ser deletado
      const testPost = posts.find(
        (p) => p.title.includes("DEBUG") || p.title.includes("TEST")
      );

      if (!testPost) {
        addDebugLine("❌ Nenhum post de teste encontrado para deletar");
        addDebugLine(
          "💡 Execute o teste de update primeiro para criar um post de teste"
        );
        return;
      }

      addDebugLine(`🗑️ Deletando post ID=${testPost.id} (${testPost.title})`);

      const result = await deletePost(testPost.id);
      addDebugLine(
        `✅ deletePost executado com sucesso: ${JSON.stringify(result)}`
      );

      // Verificar se a exclusão foi persistida
      const remainingPosts = await fetchPosts();
      const deletedPost = remainingPosts.find((p) => p.id === testPost.id);

      if (!deletedPost) {
        addDebugLine("✅ Exclusão confirmada no banco de dados");
      } else {
        addDebugLine("❌ Post ainda existe no banco após delete");
      }
    } catch (error) {
      addDebugLine(`❌ Erro no deletePost: ${error}`);
    } finally {
      setLoading(false);
    }
  };

  const clearDebug = () => {
    setDebugOutput([]);
  };

  return (
    <div className="bg-gray-100 p-6 rounded-lg mt-6">
      <h3 className="text-lg font-semibold mb-4">🐛 Debug Panel - Posts</h3>

      <div className="flex gap-2 mb-4">
        <Button onClick={testFetchPosts} disabled={loading} variant="outline">
          Test Fetch
        </Button>
        <Button onClick={testDeletePost} disabled={loading} variant="outline">
          Test Delete
        </Button>
        <Button onClick={clearDebug} disabled={loading} variant="secondary">
          Clear
        </Button>
      </div>

      <div className="bg-black text-green-400 p-4 rounded font-mono text-sm max-h-96 overflow-y-auto">
        {debugOutput.length === 0 ? (
          <p className="text-gray-500">
            Clique nos botões acima para testar as operações de posts...
          </p>
        ) : (
          debugOutput.map((line, index) => (
            <div key={index} className="mb-1">
              {line}
            </div>
          ))
        )}
      </div>
    </div>
  );
};
