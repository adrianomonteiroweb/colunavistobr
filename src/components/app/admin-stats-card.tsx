import { ReactNode } from "react";
import { Card, CardContent } from "@/components/ui/card";

export type AdminStatsCardProps = {
  icon: ReactNode;
  label: string;
  value: number | string;
  iconClassName?: string;
  valueClassName?: string;
  labelClassName?: string;
};

const AdminStatsCard = ({
  icon,
  label,
  value,
  iconClassName = "",
  valueClassName = "",
  labelClassName = "",
}: AdminStatsCardProps): React.ReactElement => (
  <Card>
    <CardContent className="p-3 md:p-6">
      <div className="flex flex-col md:flex-row md:items-center">
        <span className={iconClassName + " mb-2 md:mb-0 md:mr-3"}>{icon}</span>
        <div>
          <p
            className={
              "text-xs md:text-sm font-medium text-gray-600 " + labelClassName
            }
          >
            {label}
          </p>
          <p
            className={
              "text-lg md:text-2xl font-bold text-gray-900 " + valueClassName
            }
          >
            {value}
          </p>
        </div>
      </div>
    </CardContent>
  </Card>
);

export default AdminStatsCard;
