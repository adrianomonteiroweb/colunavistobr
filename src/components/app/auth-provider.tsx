"use client";

import type React from "react";
import { createContext, useContext, useEffect, useState } from "react";

import { createSession, destroySession, getSession } from "@/actions/auth";

interface AuthContextType {
  user: any | null;
  login: (email: string, password: string) => Promise<boolean>;
  logout: () => Promise<void>;
  isLoading: boolean;
}

const AuthContext = createContext<AuthContextType | undefined>(undefined);

export function AuthProvider({ children }: { children: React.ReactNode }) {
  const [user, setUser] = useState<any | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const loadSession = async () => {
      setIsLoading(true);
      const session: any = await getSession();

      if (session?.payload) {
        setUser(session.payload as any);
      } else {
        setUser(null);
      }
      setIsLoading(false);
    };

    loadSession();
  }, []);

  const login = async (email: string, password: string): Promise<boolean> => {
    setIsLoading(true);
    const res = await createSession(email, password);

    if (res.status === 200) {
      const session: any = await getSession();

      if (session?.payload) {
        setUser(session.payload as any);
        setIsLoading(false);
        return true;
      }
    }

    setIsLoading(false);
    return false;
  };

  const logout = async () => {
    setIsLoading(true);
    await destroySession();
    setUser(null);
    setIsLoading(false);
  };

  return (
    <AuthContext.Provider value={{ user, login, logout, isLoading }}>
      {children}
    </AuthContext.Provider>
  );
}

export function useAuth() {
  const context = useContext(AuthContext);

  if (context === undefined) {
    throw new Error("useAuth must be used within an AuthProvider");
  }
  return context;
}
