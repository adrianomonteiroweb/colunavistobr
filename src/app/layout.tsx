import type { Metadata } from "next";
import { Geist, Geist_Mono } from "next/font/google";
import "./globals.css";
import { AuthProvider } from "@/components/app/auth-provider";
import { ThemeProvider } from "@/components/theme-provider";
import { getInitialThemePalette } from "@/app/theme-palette.server";

const geistSans = Geist({
  variable: "--font-geist-sans",
  subsets: ["latin"],
});

const geistMono = Geist_Mono({
  variable: "--font-geist-mono",
  subsets: ["latin"],
});

export const metadata: Metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default async function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  const initialPalette = await getInitialThemePalette();
  return (
    <html lang="en">
      <body
        className={`${geistSans.variable} ${geistMono.variable} antialiased`}
      >
        <ThemeProvider initialPalette={initialPalette}>
          <AuthProvider>{children}</AuthProvider>
        </ThemeProvider>
      </body>
    </html>
  );
}
