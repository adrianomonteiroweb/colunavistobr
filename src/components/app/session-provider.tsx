"use client";

import type React from "react";

import { useRouter } from "next/navigation";
import { useEffect } from "react";
import { Loader2 } from "lucide-react";

import { useAuth } from "./auth-provider";

interface SessionProtectorProps {
  children: React.ReactNode;
  requireAuth?: boolean;
  redirectTo?: string;
}

export function SessionProtector({
  children,
  requireAuth = true,
  redirectTo = "/signin",
}: SessionProtectorProps) {
  const { user, isLoading } = useAuth();
  const router = useRouter();

  useEffect(() => {
    if (!isLoading) {
      if (requireAuth && !user) {
        router.push(redirectTo);
      } else if (!requireAuth && user) {
        router.push("/dashboard");
      }
    }
  }, [user, isLoading, requireAuth, redirectTo, router]);

  if (isLoading) {
    return (
      <div className="flex items-center justify-center min-h-screen">
        <Loader2 className="h-8 w-8 animate-spin" />
      </div>
    );
  }

  if (requireAuth && !user) {
    return null;
  }

  if (!requireAuth && user) {
    return null;
  }

  return <>{children}</>;
}
