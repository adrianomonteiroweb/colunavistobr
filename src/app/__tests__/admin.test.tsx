import { render, screen } from "@testing-library/react";
import { AuthProvider } from "@/components/app/auth-provider";
import type { ReactNode } from "react";
import AdminPage from "@/app/admin/page";

const MockAuthProvider = ({ children }: { children: ReactNode }) => (
  <AuthProvider>{children}</AuthProvider>
);

jest.mock("@/actions/auth", () => ({
  getSession: async () => ({
    payload: { id: 1, name: "Admin", role: "admin" },
  }),
  createSession: async () => ({ status: 200 }),
  destroySession: async () => {},
}));

describe("Admin Page", () => {
  it("should render the Admin title", async () => {
    render(
      <MockAuthProvider>
        <AdminPage />
      </MockAuthProvider>
    );
    expect(
      await screen.findByText(/painel do administrador/i)
    ).toBeInTheDocument();
  });

  it("should match snapshot", async () => {
    const { asFragment } = render(
      <MockAuthProvider>
        <AdminPage />
      </MockAuthProvider>
    );
    await screen.findByText(/painel do administrador/i);
    expect(asFragment()).toMatchSnapshot();
  });
});
