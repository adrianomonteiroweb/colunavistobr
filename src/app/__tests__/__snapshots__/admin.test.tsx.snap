// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Page should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 md:p-8"
  >
    <div
      class="text-center text-gray-500 py-12"
    >
      Painel do Administrador em construção.
    </div>
  </div>
</DocumentFragment>
`;
